### DISCUSSION TITLE
multiple relay used HomeAssistant Autodiscovery on openhab only one relay

### DISCUSSION THREAD

**Discussion by hmjvaline** [arendst/Tasmota#13771] [Q&A] [answered]
### PROBLEM DESCRIPTION
_A clear and concise description of what the problem is._
multiple relay used HomeAssistant Autodiscovery on openhab only one relay show
after setoption19 1,openhab has autodiscovery tasmota to auto create channel by tamosta template gpio setting.
but only relay1 show in openhab channel name:  state, no other relay channel
setoption26  1 is used for to solve the problem ,but no work
after setoption19 1, i find   relay2 mqtt name is empty ,i feel this is problem
becauce relay1 is tasmota_A2E12F ( <%topic%>)
please help, thank
MQT: homeassistant/switch/A2E12F_RL_2/config = {"name":"","stat_t":"tele/tasmota_A2E12F/STATE","avty_t":"tele/tasmota_A2E12F/LWT","pl_avail":"Online","pl_not_avail":"Offline","cmd_t":"cmnd/tasmota_A2E12F/POWER2","pl_off":"OFF","pl_on":"ON","val_tpl":"{{value_json.POWER2}}","uniq_id":"A2E12F_RL_2","dev":{"ids":["A2E12F"]}} (已保留)

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [ ] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [ ] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [ ] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [ ] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:
stat/tasmota_A2E12F/STATUS0 = {"Status":{"Module":18,"DeviceName":"tasmota_A2E12F","FriendlyName":["tasmota_A2E12F",""],"Topic":"tasmota_A2E12F","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"stat/%topic%/SWITCH1","SwitchMode":[1,1,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0},"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"0T00:44:14","StartupUTC":"2021-11-22T14:37:40","Sleep":50,"CfgHolder":4617,"BootCount":74,"BCResetTime":"2021-11-17T07:08:49","SaveCount":392,"SaveAddress":"F4000"},"StatusFWR":{"Version":"10.0.0(tasmota)","BuildDateTime":"2021-11-13T07:33:33","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8266EX","CR":"404/699"},"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["project","Recovery"],"TelePeriod":300,"Resolution":"558180C0","SetOption":["000A8009","2805C8000100060000005A0A000000000000","00800080","00006000","00004000"]},"StatusMEM":{"ProgramSize":775,"Free":228,"Heap":20,"ProgramFlashSize":4096,"FlashSize":4096,"FlashChipId":"1640D8","FlashFrequency":40,"FlashMode":3,"Features":["00000404","87FAC7CF","043E97A1","000000CF","011013C0","C000F981","00004024","00033000","00000020"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,13,16,18,19,20,21,22,24,26,27,29,30,35,37,45","Sensors":"1,2,3,4,5,6,40,80"},"StatusNET":{"Hostname":"tasmota-A2E12F-0303","IPAddress":"10.1.3.4","Gateway":"10.1.0.254","Subnetmask":"255.255.248.0","DNSServer1":"10.0.0.112","DNSServer2":"10.255.4.1","Mac":"84:CC:A8:A2:E1:2F","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0},"StatusMQT":{"MqttHost":"203.71.236.250","MqttPort":1883,"MqttClientMask":"DVES_%06X","MqttClient":"DVES_A2E12F","MqttUser":"DVES_USER","MqttCount":1,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4},"StatusTIM":{"UTC":"2021-11-22T15:21:54","Local":"2021-11-22T16:21:54","StartDST":"2021-03-28T02:00:00","EndDST":"2021-10-31T03:00:00","Timezone":"+01:00","Sunrise":"08:10","Sunset":"17:02"},"StatusSNS":{"Time":"2021-11-22T16:21:54","Switch1":"ON","ANALOG":{"A0":728}},"StatusSTS":{"Time":"2021-11-22T16:21:54","Uptime":"0T00:44:14","UptimeSec":2654,"Heap":19,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":1,"POWER1":"OFF","POWER2":"OFF","Wifi":{"AP":1,"SSId":"project","BSSId":"24:DE:C6:76:98:82","Channel":11,"Mode":"11n","RSSI":64,"Signal":-68,"LinkCount":1,"Downtime":"0T00:00:05"}}}

```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:
16:23:39.795 MQT: stat/tasmota_A2E12F/RESULT = {"SetOption19":"ON"}
16:23:39.882 MQT: homeassistant/device_automation/A2E12F_SW_1_TOGGLE/config =  (已保留)
16:23:39.886 MQT: homeassistant/device_automation/A2E12F_SW_1_HOLD/config =  (已保留)
16:23:39.893 MQT: homeassistant/binary_sensor/A2E12F_SW_1/config = {"name":"tasmota_A2E12F Switch1","stat_t":"stat/tasmota_A2E12F/SWITCH1","val_tpl":"{{value_json.STATE}}","frc_upd":true,"pl_on":"ON","pl_off":"OFF","uniq_id":"A2E12F_SW_1","dev":{"ids":["A2E12F"]},"avty_t":"tele/tasmota_A2E12F/LWT","pl_avail":"Online","pl_not_avail":"Offline"} (已保留)
16:23:39.951 MQT: homeassistant/sensor/A2E12F_ANALOG_A0/config = {"name":"tasmota_A2E12F ANALOG A0","stat_t":"tele/tasmota_A2E12F/SENSOR","avty_t":"tele/tasmota_A2E12F/LWT","pl_avail":"Online","pl_not_avail":"Offline","uniq_id":"A2E12F_ANALOG_A0","dev":{"ids":["A2E12F"]},"unit_of_meas":" ","ic":"mdi:eye","frc_upd":true,"val_tpl":"{{value_json['ANALOG']['A0']}}"} (已保留)
16:23:39.959 MQT: homeassistant/switch/A2E12F_RL_1/config = {"name":"tasmota_A2E12F","stat_t":"tele/tasmota_A2E12F/STATE","avty_t":"tele/tasmota_A2E12F/LWT","pl_avail":"Online","pl_not_avail":"Offline","cmd_t":"cmnd/tasmota_A2E12F/POWER1","pl_off":"OFF","pl_on":"ON","val_tpl":"{{value_json.POWER1}}","uniq_id":"A2E12F_RL_1","dev":{"ids":["A2E12F"]}} (已保留)
16:23:39.963 MQT: homeassistant/switch/A2E12F_RL_2/config = {"name":"","stat_t":"tele/tasmota_A2E12F/STATE","avty_t":"tele/tasmota_A2E12F/LWT","pl_avail":"Online","pl_not_avail":"Offline","cmd_t":"cmnd/tasmota_A2E12F/POWER2","pl_off":"OFF","pl_on":"ON","val_tpl":"{{value_json.POWER2}}","uniq_id":"A2E12F_RL_2","dev":{"ids":["A2E12F"]}} (已保留)
16:23:39.976 MQT: homeassistant/sensor/A2E12F_status/config = {"name":"tasmota_A2E12F status","stat_t":"tele/tasmota_A2E12F/HASS_STATE","avty_t":"tele/tasmota_A2E12F/LWT","pl_avail":"Online","pl_not_avail":"Offline","json_attr_t":"tele/tasmota_A2E12F/HASS_STATE","unit_of_meas":"%","val_tpl":"{{value_json['RSSI']}}","ic":"mdi:information-outline","uniq_id":"A2E12F_status","dev":{"ids":["A2E12F"],"name":"tasmota_A2E12F","mdl":"Generic","sw":"10.0.0(tasmota)","mf":"Tasmota"}} (已保留)

```

### TO REPRODUCE
_Steps to reproduce the behavior:_

### EXPECTED BEHAVIOUR
_A clear and concise description of what you expected to happen._

### SCREENSHOTS
_If applicable, add screenshots to help explain your problem._

### ADDITIONAL CONTEXT
_Add any other context about the problem here._

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment, marked as answer]
Tasmota does not know what Home Automation Hub is used. If you get a different result in discovery with OpenHab it has to be solved there. Anyway `Setoption19 1` is deprecated and should not be used anymore.

**Reply by hmjvaline** [reply]
I don't know that setoption 19 is deprecated, but it is an explanation example of setoptions in the "command" document. What I want to reflect is that the value of "name" in the mqtt json of homeassistant/switch/A2E12F_RL_2/config is empty. Is this normal? It may be corrected in the future? Thank you for your reply.

**Reply by sfromis** [reply]
Well, Jason *knows* that `SetOption19 1` *is* deprecated, leading to not much attention to details of how it could be improved. And the docs page about [HA](https://tasmota.github.io/docs/Home-Assistant/) also mentions the deprecation.

Maybe it would help your case if you actually configure the 2nd FriendlyName. Just a guess.

**Reply by hmjvaline** [reply]
thx @sfromis 2nd FriendlyName and 3rd  FriendlyName Seems to be the cause of the problem，After I confirmed the two names again, the mqtt config names are normal,that is work! Although I know setoption19  is deprecated very sad,
Because I'm afraid that no one will care about the function of autodiscovery in the future

**Reply by sfromis** [reply]
Well, it is due to caring about autodiscovery (now with `SetOption19 0`) that the improved version was developed, to get around the limitations of the older method. The new format is also better for integrating with other platforms, like OpenHAB, but someone will have to do some work.

**Reply by hmjvaline** [reply]
Indeed, I have also searched in the openhab community and found that someone proposed tasmota binding, but some people think that mqtt binding can be solved, so currently I think that having an independent openhab tasmota binding seems to be a long and difficult road

**Reply by sfromis** [reply]
Going forward, using the old style "MQTT Discovery" feature strictly built for Home Assistant cannot be "solved", as it is a deprecated feature unlikely to be improved to match additions, or fixed to deal with things "not quite right".

**Reply by Jason2866** [reply]
I use OpenHab and i like BUT when jumping on a boot to use a feature developed for a other Home Automation system the OpenHab world have to accept when breaking changes are done there.
To be clear HomeAssistant changed and Tasmota did take care of that. Some day the old way will be completely removed. No one from the dev team will invest time in this dead thing.
The Tasmota devs just need to implement the new Autodiscovery.
